clang-tools-extra/clang-tidy/readability/ImplicitBoolConversionCheck.cpp

```diff
@@ -357,20 +360,22 @@ void ImplicitBoolConversionCheck::registerMatchers(MatchFinder *Finder) {
 
 void ImplicitBoolConversionCheck::check(
     const MatchFinder::MatchResult &Result) {
-
+  if(CheckConversionsToBool){
```

Comment:
Wrong, place these things should be actually put into matcher in registerMatchers method, to speed up.

You got 2 calls to addMatcher, one in line 300, other in line 339, put those (with dependences to avoid unused local variables) into if's.

Reply:
Okay Thanks I am Working on it.

Reply:
Can You please check this? I have done this now as instructed.